tests/test_command_parse.py

```diff
@@ -193,3 +203,10 @@ def test_crawlspider_no_matching_rule(self):
         )
         self.assertRegexpMatches(to_native_str(out), """# Scraped Items  -+\n\[\]""")
         self.assertIn("""Cannot find a rule that matches""", to_native_str(stderr))
+    
+    @defer.inlineCallbacks
+    def test_retry_requests_in_callbacks(self):
+        status, out, stderr = yield self.execute(
+            ['--spider', 'retryRequests'+self.spider_name, '-d', '2', self.url('/html')]
+        )
+        self.assertIn("""INFO: Closing spider (finished)""", to_native_str(stderr))
```

Comment:
This will actually be printed even if the bug wasn't fixed:

```
  File "/home/jakob/.local/lib/python3.6/site-packages/scrapy/commands/parse.py", line 169, in callback
    cb = response.meta['_callback']
  File "/home/jakob/.local/lib/python3.6/site-packages/scrapy/http/response/__init__.py", line 30, in meta
    return self.request.meta
RecursionError: maximum recursion depth exceeded while calling a Python object
2018-02-26 10:39:30 [scrapy.core.engine] INFO: Closing spider (finished)
```

Maybe you can check the output against something that is only printed when the spider run was successful / the bug is fixed?